modules/extensions/src/terrain/terrain-pass.ts

```diff
@@ -0,0 +1,64 @@
+import {withParameters} from '@luma.gl/core';
+import {Layer, Viewport, _LayersPass as LayersPass, LayersPassRenderOptions} from '@deck.gl/core';
+import type {default as TerrainCover} from './terrain-cover';
+import type {Framebuffer} from '@luma.gl/core';
+
+import GL from '@luma.gl/constants';
+
+export type TerrainPassRenderOptions = LayersPassRenderOptions;
+
+/** Renders textures used by the TerrainEffect render pass */
+export default class TerrainCoverPass extends LayersPass {
+  getRenderableLayers(viewport: Viewport, opts: TerrainPassRenderOptions): Layer[] {
```

Comment:
If methods are overrides, can we use the `override` keyword? There is a typescript option that forces us to use it, but we aren't ready to enable that in deck.
 
```suggestion
  override getRenderableLayers(viewport: Viewport, opts: TerrainPassRenderOptions): Layer[] {
```

A JSDoc on each override would be good - why are we overriding it. I don't see anything terrain specific below. If so would this be something the superclass should do?

Reply:
These are not overrides. I guess I can add it to the superclass, except the core doesn't really need this functionality.